script.js

```diff
@@ -1,12 +1,200 @@
 //You can edit ALL of the code here
+
+let allEpisodes;
+
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  root.innerText = '';
+
+  let showContainer = document.getElementById("show-container");
+  if (showContainer) {
+    showContainer.innerText = '';
+  } else {
+    showContainer = document.createElement("div");
+    showContainer.id = "show-container";
+    root.appendChild(showContainer);
+  }
+
+  const searchBox = document.createElement("input");
+  searchBox.id = "search-box";
+  searchBox.placeholder = "Search for show name";
+  searchBox.addEventListener("keyup", renderShows);
+  showContainer.appendChild(searchBox);
+
+  let showSelectorContainer = document.createElement("div");
+  showSelectorContainer.id = "show-selector-container";
+  showContainer.appendChild(showSelectorContainer);
+
+  renderShows();
 }
 
-function makePageForEpisodes(episodeList) {
+function renderShows(event) {
+  window.scrollTo(0, 0);
+  const searchTerm = event && event.target.value.toLowerCase() || "";
+
+  let show = document.getElementById("show-container");
+
+  const showSelector = document.createElement("ul");
+  getAllShows().filter(show => show.name.toLowerCase().includes(searchTerm) || show.summary.toLowerCase().includes(searchTerm)).forEach(show => {
+    const option = document.createElement("li");
+
+    const title = document.createElement("h2");
+    title.style.clear = "both";
+    const titleLink = document.createElement("a");
+    titleLink.href = `javascript:selectShow(${show.id})`;
+    titleLink.innerText = show.name;
+    title.appendChild(titleLink);
+    option.appendChild(title);
+
+    const image = document.createElement("img");
+    image.src = show.image.medium;
+    image.style.float = "left";
+    option.appendChild(image);
+
+    const description = document.createElement("span");
+    //BOO! Insecure!
+    description.innerHTML = show.summary;
+    option.appendChild(description);
+
+    // TODO: And genre/ranking/...
+
+    showSelector.appendChild(option);
+  });
+  const showSelectorContainer = document.getElementById("show-selector-container");
+  showSelectorContainer.innerText = '';
+  showSelectorContainer.appendChild(showSelector);
+}
+
+function selectShow(id) {
+  fetch(`https://api.tvmaze.com/shows/${id}/episodes`).then(resp => resp.json()).then(episodes => {
+    allEpisodes = episodes;
+
+    const root = document.getElementById("root");
+    root.innerText = '';
+
+    const backLink = document.createElement("a");
```

Comment:
I would have used addEventHandler("click", setup) here.  6 and 1/2 dozen maybe

Reply:
But the `return false;` is so ugly! (I'm not a fan of single-page applications in the first place, but I realise how unpopular that makes me these days ;D)